service/history/execution/mutable_state_task_generator.go

```diff
@@ -86,8 +87,9 @@ type (
 	}
 
 	mutableStateTaskGeneratorImpl struct {
-		domainCache cache.DomainCache
-		logger      log.Logger
+		clusterMetadata cluster.Metadata
```

Comment:
Only store current cluster name?

Reply:
I can change to currentClusterName if you have strong opinion on it. But I kinda think it's useful to keep clusterMetadata here which can be used if we want to do additional checks in the future for example if targetCluster is a valid cluster name defined in the metadata.